rtorrent: fix version hash mismatch for 0.15.1

DenzoNL opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

The rtorrent derivation specifies that this is version 0.15.1, but when you look up the [commit hash it points to](https://github.com/rakshasa/rtorrent/commit/68fdb86c723a0ae67ebaffec416af99fec41dcbc), it turns out this actually version 0.15.0.

This PR corrects that and replaces the incorrect hash with a simple version tag.

Some trackers are extremely picky about rtorrent client versions and ban 0.15.0 but specifically allow 0.15.1, so this PR can hopefully prevent some confusion and make sure that the derivation is correct.


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

thiagokokada commented on pkgs/by-name/rt/rtorrent/package.nix:
```diff
@@ -24,8 +24,8 @@ stdenv.mkDerivation {
     repo = "rtorrent";
-    rev = "68fdb86c723a0ae67ebaffec416af99fec41dcbc";
-    hash = "sha256-/GWC28LsY7GcUH+SBzi01sOWVfA1lyM0r9OdUDTYbT8=";
+    rev = "v0.15.1";
```
```suggestion
    rev = "v${version}";
```

thiagokokada replied:
You will also need to add `rec`. Or use `finalAttrs`.

thiagokokada commented:
To be honest this PR has lots of issues, it would be probably better to backport the commits from `master` instead of possible.

DenzoNL commented:
HI @thiagokokada, thanks for your review!

Backporting from master would mean bumping the version to a newer version of rtorrent that is also problematic for me. Stable has the version I want but simply points to the wrong hash.

 I'm happy to try and fix the issues you mentioned if that means this PR would still get accepted. If not let me know, then I can just make an overlay in my personal config instead :)

thiagokokada commented:
> Backporting from master would mean bumping the version to a newer version of rtorrent that is also problematic for me. Stable has the version I want but it points to a hash of a different version than what it says it is, which is what I'm trying to fix.

In general having a non-backporting commit to a stable branch is the exception and is only done if there is a good reason. Pointing to the wrong hash revision could be the case, but in this case I would expect that we at least fix the usage of the derivation to use `finalAttrs` like we did in `master`.

DenzoNL committed:
rtorrent: fix version hash mismatch and modernize package

Backports finalAttrs pattern from master (700a1e18eb0d) while keeping
version at 0.15.1 for stable branch. This fixes the issue where the
package declared version 0.15.1 but used a commit hash pointing to
v0.15.0, causing the binary to report the wrong version.

Changes:
- Use v${finalAttrs.version} instead of hardcoded commit hash
- Switch to modern finalAttrs pattern
- Remove autoconf-archive dependency (no longer needed)
- Ensures rtorrent binary correctly reports version 0.15.1

Note: Lua support options from newer versions were not backported
as rtorrent 0.15.1 lacks --with-lua configure support.

DenzoNL commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427849`
Commit: `f432d7fea7ecf184c1b0d2152deacf860e1f67c2`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>rtorrent</li>
    <li>rtorrent.man</li>
  </ul>
</details>
